go-controller/pkg/ovn/egressip.go

```diff
@@ -2045,28 +2038,102 @@ func (oc *DefaultNetworkController) initEgressIPAllocator(node *kapi.Node) (err
 	return nil
 }
 
-// setupNodeForEgress sets up default logical router policy for every node and
-// initiates the allocator cache for the node in question, if the node has the
-// necessary annotation.
-func (oc *DefaultNetworkController) setupNodeForEgress(node *v1.Node) error {
-	if err := CreateDefaultNoRerouteNodePolicies(oc.nbClient, oc.addressSetFactory, node, oc.controllerName); err != nil {
-		oc.addEgressNodeFailed.Store(node.Name, node)
-		return err
+// reconcileNodeForEgress with respect and old and new status of a node
+func (oc *DefaultNetworkController) reconcileNodeForEgress(oldNode, newNode *v1.Node) error {
+	// Check if the node's addresses changed. If so, update LR policies.
+	if oldNode == nil || newNode == nil || util.NodeHostAddressesAnnotationChanged(oldNode, newNode) {
+		klog.Infof("Egress IP detected IP address change. Updating no re-route policies")
+		err := oc.ensureDefaultNoRerouteNodePolicies()
+		if err != nil {
+			return err
+		}
+	}
+
+	nodeEgressLabel := util.GetNodeEgressLabel()
+	var oldLabels map[string]string
+	var newLabels map[string]string
+	var isOldReady, isNewReady, isNewReachable bool
+	var nodeName string
+	if oldNode != nil {
+		oldLabels = oldNode.GetLabels()
+		isOldReady = oc.isEgressNodeReady(oldNode)
+		nodeName = oldNode.Name
+	}
+	if newNode != nil {
+		// Initialize the allocator on every update,
+		// ovnkube-node/cloud-network-config-controller will make sure to
+		// annotate the node with the egressIPConfig, but that might have
+		// happened after we processed the ADD for that object, hence keep
+		// retrying for all UPDATEs.
+		if err := oc.initEgressIPAllocator(newNode); err != nil {
+			klog.Warningf("Egress node initialization error: %v", err)
+		}
+
+		newLabels = newNode.GetLabels()
+		isNewReady = oc.isEgressNodeReady(newNode)
+		isNewReachable = oc.isEgressNodeReachable(newNode)
+		nodeName = newNode.Name
+	} else if oldNode != nil {
+		err := oc.deleteNodeForEgress(oldNode)
```

Comment:
The old code:
```
if err := h.oc.deleteNodeForEgress(node); err != nil {
	return err
}
nodeEgressLabel := util.GetNodeEgressLabel()
nodeLabels := node.GetLabels()
if _, hasEgressLabel := nodeLabels[nodeEgressLabel]; hasEgressLabel {
	if err := h.oc.deleteEgressNode(node.Name); err != nil {
		return err
	}
}
return nil
```
So `deleteNodeForEgress` was called unconditionally.
`deleteEgressNode` (confusing func names) is still called conditionally in the new code:
```
if oldHadEgressLabel && !newHasEgressLabel {
    klog.Infof("Node: %s has been un-labeled, deleting it from egress assignment", nodeName)
    return oc.deleteEgressNode(nodeName)
}
```

Reply:
Yeah, they have similar names: `deleteNodeForEgress`  vs `deleteEgressNode`

I can rename `deleteNodeForEgress` to `deleteEgressIPAllocator` because it's more less the opposite of `initEgressIPAllocator`. Thoughts?

Reply:
yes, please

Reply:
> yes, please

included in the latest push!